src/ndpp_init.F90

```diff
@@ -0,0 +1,622 @@
+            get_chi_p = .true.
+          case (SCORE_NDPP_CHI_D)
+            get_chi_d = .true.
+
+          ! Cycle through j if necessary
+          case (SCORE_SCATTER_PN, SCORE_NU_SCATTER_PN)
+            j = j + t % moment_order(j)
+            cycle SCORE_LOOP
+          case (SCORE_FLUX_YN, SCORE_TOTAL_YN, SCORE_SCATTER_YN, SCORE_NU_SCATTER_YN)
+            j = j + (t % moment_order(j) + 1)**2 - 1
+            cycle SCORE_LOOP
+        end select
+      end do SCORE_LOOP
+    end do TALLY_LOOP
+
+    ! Adjust the order so array sizing is correct
+    if (scatt_type == SCATT_TYPE_LEGENDRE) then
+      scatt_order = scatt_order + 1
+    end if
+
+  end subroutine which_data
+
+!===============================================================================
+! READ_NDPP_XML reads information from a cross_sections.xml file. This
+! file contains a listing of the ACE cross sections that may be used.
+!===============================================================================
+
+  subroutine read_ndpp_xml(scatt_type, ndpp_groups)
+
+    integer, intent(out)  :: scatt_type  ! Whether or not legendre or tabular data
+    integer, intent(out)  :: ndpp_groups ! number of groups in NDPP data
+
+    integer :: i, j, k     ! loop indices
+    logical :: file_exists ! does ndpp_lib.xml exist?
+    integer :: filetype    ! default file type
+    logical :: nuscatter   ! is nuscatter data present?
+    logical :: chi_present ! is chi data present?
+    character(MAX_WORD_LEN)  :: directory   ! directory with cross sections
+    character(MAX_LINE_LEN)  :: temp_str
+    type(TallyObject), pointer :: t => null()
+    integer :: i_filter    ! Index of filter which contains energyin or energyout
+    ! We can use the same XSListing type for our ndpp data since the NDPP
+    ! data is a subset of whats in cross_sections.xml
+    type(XsListing), pointer :: listing => null()
+    type(Node), pointer      :: doc => null()
+    type(Node), pointer      :: node_ndpp => null()
+    type(NodeList), pointer  :: node_ndpp_list => null()
+    real(8), allocatable     :: ndpp_energy_bins(:)
+    integer                  :: order       ! ndpp_lib.xml's scattering order
+
+    ! Check if ndpp_lib.xml exists
+    inquire(FILE=ndpp_lib, EXIST=file_exists)
+    if (.not. file_exists) then
+       ! Could not find ndpp_lib.xml file
+       call fatal_error("NDPP Library XML file '" // trim(ndpp_lib) // &
+            "' does not exist!")
+    end if
+
+    call write_message("Reading NDPP Library XML file...", 5)
+
+    ! Parse ndpp_lib.xml file
+    call open_xmldoc(doc, ndpp_lib)
+
+    if (check_for_node(doc, "directory")) then
+       ! Copy directory information if present
+       call get_node_value(doc, "directory", directory)
+    else
+       ! If no directory is listed in ndpp_lib.xml, by default select the
+       ! directory in which the ndpp_lib.xml file resides
+       i = index(ndpp_lib, "/", BACK=.true.)
+       directory = ndpp_lib(1:i)
+    end if
```

Comment:
There's a lot of three space indentation.  Maybe you should write a quick script to check for it.